dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi

```diff
@@ -75,4 +75,11 @@
 		label = "uart_0";
 		status = "disabled";
 	};
+
+	smc: smc {
+		compatible = "arm,sip-svc";
+		method = "smc";
+		label = "smc";
+		status = "disabled";
```

Comment:
Both sip_svc and psci is one of Arm smc service.

Arm SMCCC doc: https://developer.arm.com/documentation/den0028/b/

My understanding is PSCI is for core power management through Arm SMC reserved standard secure service calls, this will be used to support reset and SMP in Arm.

sip-svc is to provide interfaces to support "SiP Service Calls" SMC func id range which allow us to implement platform specific services.

The same for psci and sip-svc should be just the low level function that invoke smc instruction. The SMC func id ranges supported by psci and sip-svc should be different. Thus, I think we need a separated driver for sip-svc support.

Reply:
> @carlocaione I am thinking to implement this as a driver because it is using "Arm SMC SiP Service", which is to allow kernel to communicate to secure firmware (EL3) for some secure register access. Other device driver or application can use it when needed.

The problem I see is that this is not really a device driver, it is mainly a service provider (or subsystem) so having this in `drivers/` is not the correct place IMO.

@nashif @microbuilder what do you think? Do we want to discuss about this in a dev review meeting?

Reply:
I can not understand the meaning of `label = "smc"`, is it enough to use `method = "smc";`?
Also, this node looks weird, should it be something like:
```
sip0: sip_svc {
		compatible = "arm,sip-svc";
		method = "smc";
		label = "smc";
		status = "disabled";
}
```
IMO, it is a node to describe sip-svc not describe smc, right?
Correct me if I am wrong :)